crates/iota-framework/packages/iota-system/sources/validator.move

```diff
@@ -857,6 +860,27 @@ module iota_system::validator {
             return &mut validators[committee_member_index]
     }
 
+    /// If the primary address is of the form `/[ip4,ip6,dns]/{}/udp/{port}`
+    /// then it migrates into a TCP multiaddr `/[ip4,ip6,dns]/{}/tcp/{port}`.
+    /// Aborts if the current validator's `primary_address` is neither a UDP 
+    /// or TCP multiaddr.
+    public(package) fun maybe_migrate_primary_address_into_tcp(self: &mut ValidatorV1) {
```

Comment:
We also need to migrate the `next_epoch_primary_address` in the same function IMO, otherwise there could be an epoch change, and we end up with a "udp" address again in the primary address which we can't change anymore afterwards.